src/language-server/language-server.ts

```diff
@@ -59,21 +62,44 @@ export async function startLanguageServer() {
     connection.console.log('Algorand TypeScript Language Server initialized')
   })
 
-  connection.languages.diagnostics.on(async (params) => {
-    const document = documents.get(params.textDocument.uri)
-    if (document !== undefined) {
-      return {
-        kind: DocumentDiagnosticReportKind.Full,
-        items: [],
-      } satisfies DocumentDiagnosticReport
-    } else {
-      return {
-        kind: DocumentDiagnosticReportKind.Full,
-        items: [],
-      } satisfies DocumentDiagnosticReport
-    }
+  const documentChangeObservable = new Observable<TextDocumentChangeEvent<TextDocument>>((subscriber) => {
+    documents.onDidChangeContent((event) => {
+      subscriber.next(event)
+    })
   })
 
+  async function buildWorkspaceDiagnosticsMap(): Promise<Map<string, Diagnostic[]>> {
+    if (!workspaceFolder) {
+      connection.console.error('Workspace folder not set')
+
+      return new Map()
+    }
+
+    const workspacePath = URI.parse(workspaceFolder).fsPath
+    return await getWorkspaceDiagnostics(connection, workspacePath, documents)
+  }
+
+  async function sendDiagnostics(diagnosticsMap: Map<string, Diagnostic[]>) {
+    await Promise.all(
+      diagnosticsMap.entries().map(([docUri, diagnostics]) =>
+        connection.sendDiagnostics({
+          uri: docUri,
+          diagnostics: diagnostics,
+        }),
+      ),
+    )
+  }
+
+  const documentChangeSubscription = documentChangeObservable
+    .pipe(
+      debounceTime(200),
+      map(buildWorkspaceDiagnosticsMap),
+      concatMap(async (v) => sendDiagnostics(await v)),
+    )
+    .subscribe(async () => {
+      // Need a subscriber or the observable won't fire
```

Comment:
Can we change this comment to capture the reason that the subscribe handler is an empty function?